src/routes/choice-exercises/components/SingleChoiceExercise.tsx

```diff
@@ -107,36 +107,63 @@ const ChoiceExerciseScreen = ({
       return
     }
     setSelectedAnswer(clickedAnswer)
+
+    const { nthRetry } = getRetryInfoOfCurrent()
+
     const correctSelected = [correctAnswer, ...currentDocument.alternatives].find(it =>
       isAnswerEqual(it, clickedAnswer)
     )
 
     if (correctSelected !== undefined) {
       setCorrectAnswer(clickedAnswer)
-      const result: DocumentResultType = { ...currentDocument, result: SIMPLE_RESULTS.correct }
-      setResults([...results, result])
+      const result: DocumentResultType = {
+        ...currentDocument,
+        result: SIMPLE_RESULTS.correct,
+        numberOfTries: nthRetry + 1
+      }
+      updateResult(result)
     } else {
-      const result: DocumentResultType = { ...currentDocument, result: SIMPLE_RESULTS.incorrect }
-      setResults([...results, result])
+      const result: DocumentResultType = {
+        ...currentDocument,
+        result: SIMPLE_RESULTS.incorrect,
+        numberOfTries: nthRetry + 1
+      }
+      updateResult(result)
     }
     setTimeout(() => {
       setDelayPassed(true)
     }, correctAnswerDelay)
   }
 
+  const updateResult = (result: DocumentResultType): void => {
+    const indexOfCurrentResult = results.findIndex(result => result.id === currentDocument?.id)
+    indexOfCurrentResult !== -1 ? (results[indexOfCurrentResult] = result) : setResults([...results, result])
```

Comment:
You could use a map for this as well:
```suggestion
    setResults(results => results.map(result => result.id === newResult.id ? newResult : result))
```

Reply:
Not sure if a map would work, tested you solution, which does not work. 
We need to add a element to the array, if a certain condition is true, a map-function is not appropriate for this use case.

Reply:
You are right, missed that. I don't think that you should modify `results` directly like this though: `results[indexOfCurrentResult] = result`

Reply:
fixed this.